### ISSUE TITLE
dpkg install of 0.37.1-7947 not recognizing update

### ISSUE THREAD

**Issue by granitepenguin** [lbryio/lbry-desktop#3113]
**Bug**
Started up LBRY, and was prompted to update.  package was installed using 
```sudo dpkg -i /tmp/c343lF/LBRY_0.37.1.deb```
and LBRY was restarted, but it still complains there's an update.

looking at the dpkg listing, the correct version shows and was installed

```
drock@mojojojo:~$ dpkg -l | grep -i LBRY
ii  lbry      0.37.1-7947      amd64        
```
but the version listed in the About page is 0.35.3

I even uninstalled with ```dpkg -r lbry``` and tried a reinstall, with the same results.

**To Reproduce**
1. Start LBRY
2. Click on "install now" in the "Upgrade is Ready" banner
3. copy and run the "sudo dpkg -i" message to install
4. restart and get prompted to upgrade again

**Expected behavior**
running upgrade process should upgrade LBRY so I'm not prompted to upgrade again

**System Configuration**

About
A newer version of LBRY is available. 
Download now!

App	0.35.3
Daemon (lbrynet)	0.40.0
Update mailing preferences
Reward Eligible	Yes
Platform	Linux (Linux-5.2.0-3-amd64-x86_64-with-debian-bullseye-sid)

drock@mojojojo:~$ curl 'http://localhost:5279' --data '{"method":"version"}'
{
  "jsonrpc": "2.0",
  "result": {
    "build": "release",
    "desktop": "X-Cinnamon",
    "distro": {
      "codename": "n/a",
      "id": "debian",
      "like": "",
      "version": "testing/unstable",
      "version_parts": {
        "build_number": "",
        "major": "",
        "minor": ""
      }
    },
    "lbrynet_version": "0.40.0",
    "os_release": "5.2.0-3-amd64",
    "os_system": "Linux",
    "platform": "Linux-5.2.0-3-amd64-x86_64-with-debian-bullseye-sid",
    "processor": "",
    "python_version": "3.7.1",
    "torba_version": "0.5.7"
  }
}

- LBRY Installation ID:  bFmQHiwM8Js5o4ZMZznUKXcahkpjBHh4sAokCASw6SsPzAA8RC5sprkSQuBs9wV7M

**Comment by tzarebczan** [comment]
Sorry to hear about the trouble. Is it possible you have 2 versions of  LBRY installed, possibly from a flatpak? It could be that you are running that version instead of the new one being installed. 

This seems to work fine on my Linux installs.

**Comment by granitepenguin** [comment]
I don't use flatpacks, and I am manually running the binary from the terminal.

I think what's happening is the lbrynet daemon does not shut down as part of the upgrade process.  I just noticed that when I closed the client, the system tray for the daemon was still there (I was clued in by the fact closing the client left the video I was watching running and producing sound).  When I quit the daemon too and relaunched, everything is working as expected now.

```
About
Your LBRY app is up to date.

App	0.37.1
Daemon (lbrynet)	0.43.3
Reward Eligible	Yes
Platform	Linux (Linux-5.2.0-3-amd64-x86_64-with-debian-bullseye-sid)
Installation ID	bFmQHiwM8Js5o4ZMZznUKXcahkpjBHh4sAokCASw6SsPzAA8RC5sprkSQuBs9wV7M
```

So maybe this should be listed as a bug with the update process that it needs to cleanly shut everything down (client and daemon), which it apparently is not doing on my system.

**Comment by tzarebczan** [comment]
Ah, so you were not closing the app before installing with dpkg? Simply closing it minimizes it to try currently. 

What happens when you try to use the button to install the upgrade? That should shut down the app and open up the software center. This may not work on all distros. We may need to adjust the copy on the upgrade prompt to tell users to quit the app fully (ctrl-Q).

**Comment by granitepenguin** [comment]
I tried the button first, which didn't work for me.  What it does is opens the GUI package manager which does run, but it doesn't appear to shut everything down first.  I went to using dpkg because I ran into the "it didn't upgrade" issue.

**Comment by tzarebczan** [comment]
Clicking the button is what is supposed to shut things down. Is your app shutting down if you exit from the tray / press Ctrl Q? If not, send your log file over to hello@lbry.com (see help page in app).

**Comment by granitepenguin** [comment]
yes, both closing from the tray and using ctrl-Q work as expected.

**Comment by granitepenguin** [comment]
I can also see that the initial install via the GUI did not work.  You can tell by the message in the dpkg output that followed the first manual attempt (note the ```Unpacking lbry (0.37.1-7947) over (0.35.3-7161) ...```)

```
drock@mojojojo:/etc/apt/sources.list.d$ sudo dpkg -i /tmp/6XdRcS/LBRY_0.37.1.deb
[sudo] password for drock: 
(Reading database ... 289496 files and directories currently installed.)
Preparing to unpack /tmp/6XdRcS/LBRY_0.37.1.deb ...
Unpacking lbry (0.37.1-7947) over (0.35.3-7161) ...
Setting up lbry (0.37.1-7947) ...
Processing triggers for mime-support (3.64) ...
Processing triggers for gnome-menus (3.32.0-1) ...
Processing triggers for desktop-file-utils (0.24-1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
```

This has been pretty consistent on my system, which is why I'm usually doing this manually.
Clicking the button to Upgrade Now downloads the .deb, and starts the install process, but something gets lost along the way with the GUI method.

It *is* possible that LBRY is getting shut down as expected, and the handoff to the GUI package manager is what's having an issue.  As I think more about this, the steps I went through are:

1. see the "needs update" message
2. click the button
3. click install in the GUI package manager (this looks like it completed, but evidence suggests it didn't)
4. started LBRY and saw it was still the old version
it's at this point where things go really wrong.  instead of 
5. (ctrl-Q LBRY), 
I "close" LBRY by just clicking the close button on the app itself and go straight to 
5. sudo dpgk -i lbry
at which point, the daemon is probably still running because of step 4 and not realizing the client is only half the application.

I'm not sure which package manager is getting opened (I rarely use them) so I'm not sure if there's a log somewhere that can give me a better clue what it did (or didn't) do, but it seems like it's the interaction at that point that's suspect.

**Comment by tzarebczan** [comment]
I just ran through this on Ubuntu and once the package manager comes up, the app is Quit automatically in the background and everything installs fine.  So it must be related to your package manager. 

We'll add some text to the manual install instructions to quit the app completely to prevent confusion. 

I think we'll get snaps going next year which should help alleviate issues with package managers. 

Will close this issue once we add the additional instructions. 

Thanks again for filing and the research! Can we show you some [appreciation](https://lbry.com/faq/appreciation)?

**Comment by granitepenguin** [comment]
sounds good.  When the update gets pushed out, I'll try to get a better look at what it's actually trying to do when the package manager runs (ie, what PM is it, does it log, etc) and keep an eye on what LBRY does in the process (does it fully shut down, etc).

thanks for the appreciation :-)

**Comment by tzarebczan** [comment]
The instructions now say to quit the app first.

**Comment by granitepenguin** [comment]
sounds great.  

I did get a better view of the issue on my system doing this update, too.  I was able to verify that LBRY is shutting itself down, and I can see what package manager is being used (GDebi 0.9.5.7+nmu3).  It's showing the details of the .deb, but when I click "Install Package," it doesn't look like GDebi is doing what it's supposed to (ie, the dialog just goes away and the updated version is not installed).  

I then proceeded to install manually with dpkg, which worked fine this time because I made sure all parts of LBRY were shut down first.

Obviously, the issue I have to address now is why GDebi is not installing properly.  That has nothing to do with LBRY, because the .deb that gets downloaded worked fine manually.

Thanks!
